.github/actions/docker/action.yaml

```diff
@@ -4,80 +4,112 @@ inputs:
-        echo "DOCKER_TAG=us-central1-docker.pkg.dev/${PROJECT_NAME}/github/${{ inputs.docker_service }}:${GIT_SHA}" >> $GITHUB_ENV
-        echo "DOCKER_FLOATING_TAG=us-central1-docker.pkg.dev/${PROJECT_NAME}/github/${{ inputs.docker_service }}:${FLOATING_TAG}" >> $GITHUB_ENV
+        # Set Docker Build Vars
+        set -x
+
+        # Set floating to either the PR number (for PRs) or the branch name (for post-submit).
+        # Images will be tagged
+        FLOATING_TAG="${{ inputs.is_pr == 'true' && github.event.pull_request.number || github.ref_name }}"
+        # Replace slashes with double underscore.
+        FLOATING_TAG="${FLOATING_TAG//\//__}"
+        # Remove url unsafe suffix from LTS branches.
+        FLOATING_TAG="${FLOATING_TAG%.1[+,-]}"
+
+        echo "DOCKER_FLOATING_TAG=${FLOATING_TAG}" >> $GITHUB_ENV
       shell: bash
-    - name: Login to Marketplace GAR
+
+    - name: Login to GAR
       run: |
-        # Need to login to Marketplace to be able to pull image metadata.
-        PROJECT_NAME=$(gcloud config get-value project)
-        METADATA="http://metadata.google.internal./computeMetadata/v1"
-        SVC_ACCT="${METADATA}/instance/service-accounts/default"
-        ACCESS_TOKEN=$(curl -H 'Metadata-Flavor: Google' ${SVC_ACCT}/token | cut -d'"' -f 4)
+        # Login to GAR
+        TOKEN_ENDPOINT="http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token"
+        ACCESS_TOKEN=$(curl -H 'Metadata-Flavor: Google' ${TOKEN_ENDPOINT} | jq -r .access_token)
+
+        # Login to GAR to be able to push images created by fork based PR workflows.
+        if [[ "${{ inputs.is_pr }}" == "true" ]]; then
+          printf ${ACCESS_TOKEN} | docker login -u oauth2accesstoken --password-stdin https://us-central1-docker.pkg.dev
+        fi
+
+        # Login to Marketplace GAR to be able to pull base images.
         printf ${ACCESS_TOKEN} | docker login -u oauth2accesstoken --password-stdin https://marketplace.gcr.io
       shell: bash
-    # Proceed with docker pull, build, tag, and push.
-    - name: Docker Pull
-      run: |
-        docker pull "${DOCKER_TAG}" || true
-        docker pull "${DOCKER_FLOATING_TAG}" || true
-        cat ${GITHUB_WORKSPACE}/cobalt/docker/unittest/Dockerfile
-      shell: bash
-    - name: Docker Build
-      run: docker compose build ${{inputs.docker_service}}
-      shell: bash
-    - name: Docker Tag
+
+    - name: Login to GitHub Docker Registry
+      uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ inputs.github_token }}
+
+    - name: Pull Existing Docker Image
+      id: pull-cached-image
+      env:
+        GITHUB_DOCKER_REGISTRY_URL: ghcr.io
+        PR_DOCKER_REGISTRY_URL: us-central1-docker.pkg.dev
+        DOCKER_SERVICE: ${{ inputs.docker_service }}
+        CONTENT_SHA: ${{ inputs.docker_content_sha }}
```

Comment:
nit: docker_content_sha? just for keeping uniformity among docker specific stuff